### ISSUE TITLE
[Bug]: Theming: Default Background Image does not use instance color

### ISSUE THREAD

**Issue by newhinton** [nextcloud/server#35002]
### ⚠️ This issue respects the following points: ⚠️

- [X] This is a **bug**, not a question or a configuration/webserver/proxy issue.
- [X] This issue is **not** already reported on Github _(I've searched it)_.
- [X] Nextcloud Server **is** up to date. See [Maintenance and Release Schedule](https://github.com/nextcloud/server/wiki/Maintenance-and-Release-Schedule) for supported versions.
- [X] Nextcloud Server **is** running on 64bit capable CPU, PHP and OS.
- [X] I agree to follow Nextcloud's [Code of Conduct](https://nextcloud.com/contribute/code-of-conduct/).

### Bug description

When the user changes the background image, the color gets changed. For the Default Image, the nextcloud-default-blue is used, NOT the instance-color set by an admin. Therefore it is not possible to get back to the "defaults".

### Steps to reproduce

1. Change Background Image away from Default
2. Change Image back to default.

### Expected behavior

The color should now be what was set in the global theming app, in my case red.
Instead it is blue.

### Installation method

Community Manual installation with Archive

### Operating system

Debian/Ubuntu

### PHP engine version

PHP 7.4

### Web server

Apache (supported)

### Database engine version

MariaDB

### Is this bug present after an update or on a fresh install?

Updated to a major version (ex. 22.2.3 to 23.0.1)

### What user-backends are you using?

- [X] Default user-backend _(database)_
- [ ] LDAP/ Active Directory
- [ ] SSO - SAML
- [ ] Other

### List of activated Apps

```shell
- theming: 2.0.1
```

**Comment by skjnldsv** [comment]
@newhinton fixed in 25.0.1, no?

**Comment by newhinton** [comment]
No it is present in 25.0.1. 

Though i am not sure if i used the beta-channel to update.

**Comment by skjnldsv** [comment]
If you went through the beta, there might have been some config loss in the background config you might have chosen before.

For me it works properly.
There is one flow that might be confusing though, can you confirm this is what you are refering to or not?
1. Admin set color to COLOR1
2. User select "plain background". User now have COLOR1 as a background
3. Admin set new color to COLOR2
4. User STILL have COLOR1 defined

If this is about that, then it's per design. "plain background" is a one-time set option.
This is not "Follow instance colors".

**Comment by skjnldsv** [comment]
> It seems a fresh install is not affected (https://github.com/nextcloud/server/commit/944be7950a0199dbc48b99dac936987b13dd0383)

Yep, this is fixed, but you have to select the "default" again.
That will clear the old saved colour and fallback to the admin theming COLOR1.

So this ticket can remain closed.
We cannot have a migration as we don't know if the saved user colour is actually done on purpose or not by the user

**Comment by newhinton** [comment]
> Yep, this is fixed, but you have to select the "default" again.

Yes i understand, but selecting default again does not use the color set by the instance-admin, it uses blue. 
I dont know if my local test instance is just a fluke, i will wait for 25.0.2 to be released to check if it is solved in my production-environment.

**Comment by newhinton** [comment]
@skjnldsv I have now updated my production-instance to 25.0.2, and the behaviour is still there.

"Default Image" does not use the proper value, even when i switch it multiple times, or change it via admin-theming.


I think the issue is here:

https://github.com/nextcloud/server/blob/49a4aa408978039e2b1ac50c78568f1a10e7d348/apps/theming/lib/ThemingDefaults.php#L232-L235

I could not confirm that this is actually happening via the occ command, but from reading the code that case should return 

```
// admin-defined primary color
$defaultColor = $this->getDefaultColorPrimary();
```
aswell.

**Comment by skjnldsv** [comment]
Thanks! I guess we got mistaken for a different one, theming got messy with 25
https://github.com/nextcloud/server/pull/35702

**Comment by szaimen** [comment]
> Thanks! I guess we got mistaken for a different one, theming got messy with 25
> #35702

Indeed this one should have landed in 25.0.3.

@newhinton can you upgrade to 25.0.3 and check if the problem is still there? Thanks! :)

**Comment by newhinton** [comment]
It does not! It seems fixed. Thanks!
